Algorithm/QCAlgorithm.Framework.cs

```diff
@@ -394,7 +398,35 @@ public void EmitInsights(params Insight[] insights)
         /// <param name="insight">The insight to be emitted</param>
         public void EmitInsights(Insight insight)
         {
-            EmitInsights(new []{insight});
+            EmitInsights(new[] { insight });
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="insights"></param>
+        /// <returns></returns>
+        private Insight[] InitializeInsights(Insight[] insights)
+        {
+            // Verify insights are valid and initialize their fields
+            var validInsights = new List<Insight>();
+            foreach (var insight in insights)
```

Comment:
Cool!
This is very sensitive to performance since it's executed for each insight, suggest we do the following:
- using a simple `for(var i = 0;..)` loop, so that we do not create the enumerator using linq.
- avoid creating the list unless there is at least 1 delisted security which we should remove, so in 99.9% of the cases we do not create a new collection at all, we return provided `Insight[]`
- let's make `OnInsightsGenerated` take an `Insight[]` and avoid the `ToArray` call he has internally

nit - missing XML docs

Reply:
Gotcha!